pkg/status/status.go

```diff
@@ -104,22 +105,26 @@ func UpdateConditionsInPlace(conds []metav1.Condition, condition metav1.Conditio
 		return false // should never happen
 	}
 
```

Comment:
we can now use `metahelper.FindStatusCondition` right? If so let's do it (in a separate commit/PR)